app/src/observables/webworker-messaging/common.tsx

```diff
@@ -0,0 +1,213 @@
+  // let scheduled = messenger
+
+  const messengers: IMessengers = {
+    next: (value: IPyodideMessage) => messenger.next(value),
+    subscribe: (func: (value: IPyodideMessage) => void) => { scheduled.subscribe(func) },
+    executeRequest: scheduled.pipe(filter((data: IPyodideMessage) => data.type === 'executeRequest')) as Observable<IExecuteRequestMessage>,
+    fileTransfer: scheduled.pipe(filter(data => data.type === 'fileTransfer')) as Observable<IFileTransferMessage>,
+    fileTransferRequest: scheduled.pipe(filter(data => data.type === 'fileTransferRequest')) as Observable<IFileTransferRequestMessage>,
+    languageServer: scheduled.pipe(filter(data => data.type === 'languageServer')) as Observable<ILanguageServerMessage>,
+    reply: scheduled.pipe(filter(data => data.type === 'reply')) as Observable<IReplyMessage>,
+    initialise: scheduled.pipe(filter(data => data.type === 'initialise')) as Observable<IInitialiseMessage>
+  }
+
+  return messengers
+}
+
+const createMessengers = () => {
+  const receiver = createBaseMessengers()
+  const sender = createBaseMessengers()
+
+  const sendMessage = (data: IPyodideData, type: any) => {
+    const uuid = createUuid();
+    const responses = receiver.reply.pipe(filter(message => message.uuid === uuid))
+
+    const message: IPyodideMessage = {
+      uuid: uuid,
+      type: type,
+      data: data
+    }
+
+    sender.next(message)
+    return responses
+  }
+
+  const sendExecuteRequest = (code: string): Observable<IReplyMessage> => {
+    const data: IExecuteRequestData = { code }
+    return sendMessage(data, 'executeRequest')
+  }
+
+  const sendLanguageServer = (data: ILanguageServerMessage): Observable<IReplyMessage> => {
+    return sendMessage(data, 'languageServer')
+  }
+
+  const sendInitialise = (): Observable<IReplyMessage> => {
+    return sendMessage({}, 'initialise')
+  }
+
+  const sendReply = (uuid: string, data: IReplyData) => {
+    const message: IReplyMessage = {
+      uuid: uuid,
+      type: 'reply',
+      data: data
+    }
+
+    sender.next(message)
+  }
+
+  const sendFileTransferRequest = (filepath: string): Observable<IFileTransferMessage> => {
+    const uuid = createUuid();
+    const responses = receiver.fileTransfer.pipe(filter(message => message.uuid === uuid))
+
+    const message: IPyodideMessage = {
+      uuid: uuid,
+      type: 'fileTransferRequest',
+      data: { filepath }
+    }
+
+    sender.next(message)
+    return responses
+  }
+
+  const sendFileTransfer = (file: ArrayBuffer, filepath: string, uuid?: string): Observable<IReplyMessage> => {
+    if (uuid === undefined) {
+      uuid = createUuid();
+    }
+    const responses = receiver.reply.pipe(filter(message => message.uuid === uuid))
+    const message: IFileTransferMessage = {
+      uuid: uuid,
+      type: 'fileTransfer',
+      data: { file, filepath },
+    }
+
+    message.transferables = [message.data.file]
+
+    sender.next(message)
+    return responses
```

Comment:
Here is the logic for a range of created functions... such as `sendFileTransfer`